### PR TITLE
Bump rust-mozjs for the safe Runtime::new (fixes #11512)

### LINKED ISSUE #11512: Check if the script thread outlives web worker threads
This may be a problem now that we use the parent runtime support.

CC @tschneidereit @nox @jdm

### PR DESCRIPTION
**Pull request by nox**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/11520)

### CODE REVIEW

**Comment by nox** [comment]
r? @Ms2ger
Cc @tschneidereit

File: components/script/script_runtime.rs [hunk 1]
```diff
@@ -97,7 +97,7 @@ impl<'a> Drop for StackRootTLS<'a> {
 #[allow(unsafe_code)]
 pub unsafe fn new_rt_and_cx(parent_rt: *mut JSRuntime) -> Runtime {
```
Comment (jdm) [inline components/script/script_runtime.rs]: Is this unused now?

Comment (nox) [inline components/script/script_runtime.rs]: Yes it is. No idea why I didn't remove that. This is fixed now.

**Comment by tschneidereit** [comment]
Thanks, this is much nicer, and will get us a lot more sharing. r=me

<details><summary><i>Previously, highfive wrote…</i></summary>
> New code was committed to pull request.
</details>


---

Reviewed 2 of 3 files at r1, 3 of 3 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/servo/servo/11520#-:-KJ5lURYZtwQyFBESZEW:1751306291)*

**Comment by nox** [comment]
Gah, failed an import... This is fixed now.

@bors-servo r=tschneidereit

**Pull request merged**